packages/eslint-plugin-testing-library-react/src/rules/prefer-expect-query-by.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import { createRule } from '../utils';
+
+interface ExtendedExpression extends TSESTree.CallExpression {
+  callee: TSESTree.MemberExpression | TSESTree.Identifier;
+}
+
+function isIdentifier(
+  node: ExtendedExpression['callee']
+): node is TSESTree.Identifier {
+  return node.type === AST_NODE_TYPES.Identifier;
+}
+
+function isMemberExpression(
+  node: ExtendedExpression['callee']
+): node is TSESTree.MemberExpression {
+  return node.type === AST_NODE_TYPES.MemberExpression;
+}
+
+function isUsingWrongQueries(node: TSESTree.Identifier) {
+  return node.name.startsWith('getBy') || node.name.startsWith('getAllBy');
+}
+
+// https://github.com/microsoft/TypeScript/issues/16069#issuecomment-369374214
+function isNotNullOrUndefined<T>(input: null | undefined | T): input is T {
+  return input != null;
+}
+
+function mapNodesForWrongGetByQuery(node: TSESTree.CallExpression) {
+  const nodeArguments = node.arguments as ExtendedExpression[];
+  return nodeArguments
+    .map(arg => {
+      if (!arg.callee) {
+        return null;
+      }
+      // Example: `expect(rendered.getBy*)`
+      if (isMemberExpression(arg.callee)) {
```

Comment:
This case is represented like this as an AST:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1110551/63999524-f8bd3280-cb03-11e9-9aed-abd9e3563faa.png">